src/components/popover/bl-popover.ts

```diff
@@ -0,0 +1,200 @@
+import { CSSResultGroup, html, LitElement, TemplateResult } from 'lit';
+import { customElement, property, query, state } from 'lit/decorators.js';
+import {
+  computePosition,
+  flip,
+  shift,
+  offset,
+  arrow,
+  inline,
+  autoUpdate,
+  size,
+  Middleware, MiddlewareState,
+} from '@floating-ui/dom';
+import { classMap } from 'lit/directives/class-map.js';
+import style from './bl-popover.css';
+import { event, EventDispatcher } from '../../utilities/event';
+
+export type Placement =
+  | 'top-start'
+  | 'top'
+  | 'top-end'
+  | 'bottom-start'
+  | 'bottom'
+  | 'bottom-end'
+  | 'left-start'
+  | 'left'
+  | 'left-end'
+  | 'right-start'
+  | 'right'
+  | 'right-end';
+
+/**
+ * @tag bl-popover
+ * @summary Baklava Popover component
+ *
+ * @property {string} placement - Sets the popover placement
+ */
+@customElement('bl-popover')
+export default class BlPopover extends LitElement {
+  static get styles(): CSSResultGroup {
+    return [style];
+  }
+
+  @query('.popover') private popover: HTMLElement;
+  @query('.arrow') private arrow: HTMLElement;
+
+  /**
+   * Sets placement of the popover
+   */
+  @property({ type: String })
+  placement: Placement = 'top';
+
+  @state() _target: string | Element;
+
+  @property({ type: Boolean })
+  shift = false;
+
+  @property({ type: Boolean, attribute: 'fit-size' })
+  fitSize = false;
+
+  @property({ type: Number })
+  offset = 10;
```

Comment:
Add jsdoc documents for those properties.

Reply:
What does jsdoc mean? Did you mean stories?